libevmasm/BlockDeduplicator.h

```diff
@@ -64,9 +64,14 @@ class BlockDeduplicator
 	/// Iterator that skips tags and skips to the end if (all branches of) the control
 	/// flow does not continue to the next instruction.
 	/// If the arguments are supplied to the constructor, replaces items on the fly.
-	struct BlockIterator: std::iterator<std::forward_iterator_tag, AssemblyItem const>
+	struct BlockIterator
 	{
 	public:
+		using iterator_category = std::forward_iterator_tag;
+		using value_type = AssemblyItem const;
+		using difference_type = long;
```

Comment:
Why `long` and not `ptrdiff_t` or something?

Reply:
Or `size_t`?

Reply:
I didn't know of `ptrdiff_t`. The compiler warning informed me the conversion was to `long` and hence I chose it. I will change this to `ptrdiff_t`.